makelib/common.mk

```diff
@@ -105,13 +105,6 @@ endif
 # Set the host's arch.
 HOSTARCH := $(shell uname -m)
 
-# Set safe architectures for supported OSes (darwin and linux).
-# Apple Silicon binaries are not widely available yet
-ifeq ($(HOSTOS),darwin)
-SAFEHOSTARCH := amd64
```

Comment:
SAFEHOSTARCH has been necessary because of dependencies in https://github.com/upbound/build/blob/master/makelib/golang.mk and https://github.com/upbound/build/blob/master/makelib/k8s_tools.mk

At the beginning of the year several tools did not offer arm64 binaries. So we would fall back to amd64, which the m1 can run via rosetta.

Reply:
Gotcha. I removed it here due to some image tagging that's happening in crossplane. I can find a different way of handling it in the scripts over there if need be.